### ISSUE TITLE
Wavelet mode failing if output_all=True

### ISSUE THREAD

**Issue by halecatherine** [lofar-astron/PyBDSF#200]
When I am running PyBDSF on a test image with imaging commands which include the atrous fitting mode, it is running fine, however if I include the output_all = True option, then it processes the image but is failing on the first wavelet mode as it is unable to find an input directory and has a filename which does not appear to agree with files saved. 

```>>> img = bdsf.process_image('J105007+304037.fits', detection_image='J105007+304037.fits', thresh_isl=4.0, thresh_pix=5.0, rms_box=(150,15), rms_map=True, mean_map='zero', ini_method='intensity', adaptive_rms_box=True, adaptive_thresh=150, rms_box_bright=(60,15), group_by_isl=False, group_tol=10.0, output_opts=True, output_all=True, atrous_do=True, atrous_jmax=4, flagging_opts=True, flag_maxsize_fwhm=0.5, advanced_opts=True, blank_limit=None, frequency=1400000000)
--> Opened 'J105007+304037.fits'
Image size .............................. : (667, 667) pixels
Number of channels ...................... : 1
Number of Stokes parameters ............. : 1
Beam shape (major, minor, pos angle) .... : (1.50000e-03, 1.50000e-03, 0.0) degrees
Frequency of image ...................... : 1400.000 MHz
Number of blank pixels .................. : 0 (0.0%)
Flux from sum of (non-blank) pixels ..... : 0.325 Jy
--> Calculating background rms and mean images
--> Using adaptive scaling of rms_box
Using user-specified rms_box ............ : (60, 15) pixels (small scale)
Using user-specified rms_box ............ : (150, 15) pixels (large scale)
Number of sources using small scale ..... : 1
--> Using 2D map for background rms
--> Using constant background mean
Min/max values of background rms map .... : (1.33e-04, 1.91e-04) Jy/beam
Value of background mean ................ : 0.0 Jy/beam
--> Expected 5-sigma-clipped false detection rate < fdr_ratio
--> Using sigma-clipping ('hard') thresholding
Minimum number of pixels per island ..... : 6

--> Determining islands from detection image
--> Opened 'J105007+304037.fits'
Image size .............................. : (667, 667) pixels
Number of channels ...................... : 1
Number of Stokes parameters ............. : 1
Beam shape (major, minor, pos angle) .... : (1.50000e-03, 1.50000e-03, 0.0) degrees
Frequency of image ...................... : 1400.000 MHz
Number of blank pixels .................. : 0 (0.0%)
--> Calculating background rms and mean images
--> Using adaptive scaling of rms_box
Using user-specified rms_box ............ : (60, 15) pixels (small scale)
Using user-specified rms_box ............ : (150, 15) pixels (large scale)
Number of sources using small scale ..... : 1
--> Using 2D map for background rms
--> Using constant background mean
Min/max values of background rms map .... : (1.33e-04, 1.91e-04) Jy/beam
Value of background mean ................ : 0.0 Jy/beam
--> Expected 5-sigma-clipped false detection rate < fdr_ratio
--> Using sigma-clipping ('hard') thresholding
Minimum number of pixels per island ..... : 6
Number of islands found ................. : 8
> /home/chale/.local/lib/python3.10/site-packages/bdsf/functions.py(1432)write_image_to_file()
-> outdir = img.indir
(Pdb) print(img.indir)
./
(Pdb) c
> /home/chale/.local/lib/python3.10/site-packages/bdsf/functions.py(1432)write_image_to_file()
-> outdir = img.indir
(Pdb) c

--> Continuing processing using primary image
Fitting islands with Gaussians .......... : [========] 8/8
Total number of Gaussians fit to image .. : 20
Total flux density in model ............. : 0.359 Jy

--> Wavelet scale #1
--> Calculating background rms and mean images
--> Using adaptive scaling of rms_box
Using user-specified rms_box ............ : (60, 20) pixels (small scale)
Using user-specified rms_box ............ : (150, 50) pixels (large scale)
Number of sources using small scale ..... : 1
--> Using 2D map for background rms
--> Using constant background mean
Min/max values of background rms map .... : (8.33e-05, 1.46e-04) Jy/beam
Value of background mean ................ : 0.0 Jy/beam
Number of islands found ................. : 0
> /home/chale/.local/lib/python3.10/site-packages/bdsf/functions.py(1432)write_image_to_file()
-> outdir = img.indir
(Pdb) img.indir
*** AttributeError: 'Image' object has no attribute 'indir'
(Pdb) img.filename
'J105007+304037.fits.atrous.w1'
```

whereas if I look at the path where the files are created:
``` ls J105007+304037_pybdsf/wavelet/ ```

I get:
```background  J105007+304037.pybdsf.atrous.w1.fits  model  residual```

this is using the add_detection_opts branch

**Comment by darafferty** [comment]
Thanks for reporting this issue (and including the traceback!). The problem with the missing `indir` attribute should be fixed now on the `add_detection_opts` branch (for some reason the problem does not seem to occur on the `master` branch). Please report back if you still have issues.

**Comment by darafferty** [comment]
This issue should be resolved with PR #209. If not, please reopen.
